// @novu/node/build/module/lib/topics/topics.js
import { WithHttp } from '../novu.interface';
export class Topics extends WithHttp {
    async addSubscribers(topicKey, data) {
        return await this.http.post(`/topics/${topicKey}/subscribers`, {
            ...data,
        });
    }
    async removeSubscribers(topicKey, data) {
        return await this.http.post(`/topics/${topicKey}/subscribers/removal`, {
            ...data,
        });
    }
    async list(data) {
        return await this.http.get(`/topics`, {
            params: {
                page: data.page,
                ...(data?.pageSize && { pageSize: data.pageSize }),
            },
        });
    }
    async get(topicKey) {
        return await this.http.get(`/topics/${topicKey}`);
    }
    async getSubscriber(topicKey, externalSubscriberId) {
        return await this.http.get(`/topics/${topicKey}/subscribers/${externalSubscriberId}`);
    }
    async rename(topicKey, newName) {
        return await this.http.patch(`/topics/${topicKey}`, {
            name: newName,
        });
    }
    async create(data) {
        return await this.http.post(`/topics`, {
            key: data.key,
            name: data.name,
        });
    }
}
